### DISCUSSION TITLE
why return notEOF error  with error is nil?

### DISCUSSION THREAD

**Discussion by zhuliquan** [alecthomas/participle#355] [General]
https://github.com/alecthomas/participle/blob/220c1db005e6074a55b42cfffe1dc90e129cef4d/nodes.go#L495

I am confused why return notEOF error  with error is nil?

**Comment by alecthomas** [comment]
This is the negation node which matches anything _but_ what is defined. In this case `notEOF` is a token that did not satisfy this requirement and is an error.
